airflow/contrib/hooks/spark_submit_hook.py

```diff
@@ -255,12 +254,11 @@ def _process_log(self, itr):
 
     def on_kill(self):
         if self._sp and self._sp.poll() is None:
-            logging.info('Sending kill signal to spark-submit')
-            self.sp.kill()
+            logging.info('Sending kill signal to %s' % self._connection['spark_binary'])
+            self._sp.kill()
 
             if self._yarn_application_id:
                 logging.info('Killing application on YARN')
```

Comment:
Sorry for the delay.

I've read a bit about subprocess and I think the `.wait()` should be avoided, if possible. As the [documentation warns](https://docs.python.org/2/library/subprocess.html#subprocess.Popen.wait), the process can deadlock if the output is too big (see also [this blogpost](https://thraxil.org/users/anders/posts/2008/03/13/Subprocess-Hanging-PIPE-is-your-enemy/)). Using the `check_call` per your suggestion would be a good idea: the process won't deadlock, there won't be an error (assuming this is desirable), but it's still logged.